### PR TITLE
complete first iteration

### PR DESCRIPTION
**Pull request by sofialapteva**

### CODE REVIEW

File: lib/homework1.ex [hunk 1]
```diff
@@ -1,18 +1,50 @@
 defmodule Homework1 do
   @moduledoc """
-  Documentation for `Homework1`.
+  This module implements the simple calculation function.
+  You can use the Homework1.calculate/1 function to add, remove, multiply and divide.
```
Comment (am-kantox) [inline lib/homework1.ex:4]:
```suggestion
  You can use the `Homework1.calculate/1` function to add, remove, multiply and divide.
```
That way `ex_doc` would _link_ it in the generated documentation automagically.

File: lib/homework1.ex [hunk 2]
```diff
@@ -1,18 +1,50 @@
...
+  @spec calculate(String.t()) :: number()
+  def calculate(expression) when is_binary(expression) do
+    expression
+    |> String.replace(" ", "")
+    |> Float.parse()
+    |> compute()
+  end
+
+  @spec compute({Float.t(), String.t()}) :: number()
+  defp compute({a, "+" <> rest}) do
```
Comment (am-kantox) [inline lib/homework1.ex:31]: Smart :)

You might even do recursively call `compute/1` on the rest coming from the _next_ `Float.parse/1` to support longer expressions with left-to-right order of evaluation.

Somewhat like
```elixir
defp compute({result, ""}), do: result
defp compute({result, "+" <> rest}) do
  {b, rest} = Float.parse(rest)
  compute({a + b, rest})
end
```

File: lib/homework1.ex [hunk 3]
```diff
@@ -1,18 +1,50 @@
...
+  end
+
+  defp compute({a, "*" <> rest}) do
+    {b, _} = Float.parse(rest)
+    a * b
+  end
+
+  defp compute({a, "/" <> rest}) do
+    {b, _} = Float.parse(rest)
+    a / b
```
Comment (am-kantox) [inline lib/homework1.ex:48]: Calculators usually don’t blow up in fire when you try to divide by zero. I’d handle the case of `"10 / 0"` explicitly, showing a message instead of raising an exception here.

File: lib/homework1.ex [hunk 4]
```diff
@@ -1,18 +1,50 @@
...
+
+  @spec calculate(String.t()) :: number()
+  def calculate(expression) when is_binary(expression) do
+    expression
+    |> String.replace(" ", "")
+    |> Float.parse()
+    |> compute()
+  end
+
+  @spec compute({Float.t(), String.t()}) :: number()
```
Comment (am-kantox) [inline lib/homework1.ex:30]: Don’t write stuff just for the sake of writing more stuff.

To my best knowledge, there is no [`t:Float.t/0`](https://hexdocs.pm/elixir/Float.html#types).  
```suggestion
  @spec compute({float(), String.t()}) :: number()
```

File: mix.exs [hunk 5]
```diff
@@ -21,8 +21,8 @@ defmodule Homework1.MixProject do
   # Run "mix help deps" to learn about dependencies.
   defp deps do
     [
-      # {:dep_from_hexpm, "~> 0.3.0"},
-      # {:dep_from_git, git: "https://github.com/elixir-lang/my_dep.git", tag: "0.1.0"}
+      {:dialyxir, "~> 1.4", only: [:dev, :test], runtime: false},
```
Comment (am-kantox) [inline mix.exs:24]: I love dialyzer, but it should fail if ran :)

File: test/homework1_test.exs [hunk 6]
```diff
@@ -2,7 +2,27 @@ defmodule Homework1Test do
...
+  test "multiply" do
+    assert Homework1.calculate("2*5") === 10.0
+    assert Homework1.calculate("-2*5") === -10.0
+    assert Homework1.calculate("-2*0") === 0.0
+  end
+
+  test "divide" do
+    assert Homework1.calculate("2/5") === 0.4
+    assert Homework1.calculate("-2.5/5") === -0.5
+    assert Homework1.calculate("-2/1") === -2.0
```
Comment (am-kantox) [inline test/homework1_test.exs:26]: One should test the obvious corner case here, instead of coquettishly pretending that it does not exist. You’ve chosen to raise, ok, then test it, we have [`assert_raise/2`](https://hexdocs.pm/ex_unit/ExUnit.Assertions.html#assert_raise/2) for that.

**Review by am-kantox — APPROVED** [review]
:zap: :droplet: :zap: :gem: :zap: :blue_heart: :zap:

I like it because it’s not straightforward, and the solution is not standard.
